staging/src/k8s.io/csi-translation-lib/plugins/in_tree_volume.go

```diff
@@ -270,7 +271,7 @@ type regionTopologyHandler func(pv *v1.PersistentVolume) error
 // 1. Replace all CSI topology to Kubernetes Zone topology label
 // 2. Process and generate region topology if a regionTopologyHandler is passed
 // 3. Add Kubernetes Topology labels(zone) if they do not exist
-func translateTopologyFromCSIToInTree(pv *v1.PersistentVolume, csiTopologyKey string, regionTopologyHandler regionTopologyHandler) error {
+func translateTopologyFromCSIToInTree(pv *v1.PersistentVolume, csiTopologyKey string, regionTopologyHandler regionTopologyHandlerFn, regionParser regionParserFn) error {
```

Comment:
I tried to make regionParser part of regionTopologyHandler, but couldn't find a simple way. Open to suggestions :)

Reply:
What about remove the type regionTopologyHandlerFn, I think the `regionTopologyHandler()` should be generic enough to be shared?
And then the code will be like:
```
func translateTopologyFromCSIToInTree(pv *v1.PersistentVolume, csiTopologyKey string, regionParser regionParserFn) error {
```

Reply:
I actually did that change but then reverted it back because of [this line](https://github.com/kubernetes/kubernetes/pull/99130/files#diff-40f96191a7878d92ff0d264697956ff4aeb25da46763ec3367350c8c77a05598R285) specifically. If we're OK with using the `regionParser` for the nil check, should be a trivial change.

Reply:
I think it should be okay